src/app/student/student.service.ts

```diff
@@ -108,14 +108,17 @@ export class StudentService {
   }
 
   getSecurityQuestion(username) {
-    const params = new HttpParams().set('username', username);
+    let params = new HttpParams();
+    params = params.set('username', username);
+
     return this.http.get<any>(this.getSecurityQuestionUrl, { params: params });
   }
 
-  checkSecurityAnswer(username, answer) {
+  checkSecurityAnswer(username, answer, token) {
     const headers = new HttpHeaders().set('Content-Type', 'application/x-www-form-urlencoded');
     let params = new HttpParams().set('username', username);
     params = params.set('answer', answer);
+    params = params.set('token', token);
     return this.http.post<any>(this.checkSecurityAnswerUrl, params, { headers: headers });
   }
```

Comment:
I think you can undo the changes in this file because they are unrelated to this issue. It looks like these changes were made for adding recaptcha to the forgot student password form.